accumulator/src/light.rs

```diff
@@ -64,53 +60,46 @@ impl IncrementalMerkle {
     }
 
     /// Get the leading-edge branch.
-    pub fn branch(&self) -> &[H256; TREE_DEPTH] {
+    pub fn branch(&self) -> &[H256; N] {
         &self.branch
     }
 
     /// Calculate the root of a branch for incremental given the index
-    pub fn branch_root(item: H256, branch: [H256; TREE_DEPTH], index: usize) -> H256 {
+    pub fn branch_root(item: H256, branch: [H256; N], index: usize) -> H256 {
         merkle_root_from_branch(item, &branch, 32, index)
     }
 
     /// Verify a incremental merkle proof of inclusion
-    pub fn verify(&self, proof: &Proof) -> bool {
+    pub fn verify(&self, proof: &Proof<N>) -> bool {
         let computed = IncrementalMerkle::branch_root(proof.leaf, proof.path, proof.index as usize);
         computed == self.root()
     }
 }
 
-#[cfg(test)]
-mod test {
-    use ethers::utils::hash_message;
-
-    use super::*;
-    use crate::test_utils;
-
-    #[test]
-    fn it_computes_branch_roots() {
-        let test_cases = test_utils::load_merkle_test_json();
-
-        for test_case in test_cases.iter() {
-            let mut tree = IncrementalMerkle::default();
-
-            // insert the leaves
-            for leaf in test_case.leaves.iter() {
-                let hashed_leaf = hash_message(leaf);
-                tree.ingest(hashed_leaf);
-            }
-
-            // assert the tree has the proper leaf count
-            assert_eq!(tree.count(), test_case.leaves.len());
-
-            // assert the tree generates the proper root
-            let root = tree.root(); // root is type H256
-            assert_eq!(root, test_case.expected_root);
-
-            for n in 0..test_case.leaves.len() {
-                // check that the tree can verify the proof for this leaf
-                assert!(tree.verify(&test_case.proofs[n]));
-            }
-        }
-    }
-}
+// #[cfg(test)]
+// mod test {
+//     use ethers::utils::hash_message;
+//     use super::*;
+//     use crate::test_utils;
```

Comment:
Seems like issue here is getting `test_utils` imported because its in `nomad-core`. One long winded but proper solution would be to create `nomad-core-test` and put test_utils in `nomad-core-test`. 

`nomad-test` should be renamed `nomad-base-test` anyways since all its utils are for agent usage (db setup/teardown + mock contract interfaces).

Reply:
just not worrying about it for now 😅

Reply:
Lol ok one more in the chores pile 😅: https://github.com/nomad-xyz/rust/issues/48